Allow loading dependencies by name (#1594)
uxlfoundation/oneTBB: CMakeLists.txt

option(TBB_CPF "Enable preview features of the library" OFF)
option(TBB_FIND_PACKAGE "Enable search for external oneTBB using find_package instead of build from sources" OFF)
option(TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH "Disable HWLOC automatic search by pkg-config tool" ${CMAKE_CROSSCOMPILING})

> option(TBB_VERIFY_DEPENDENCY_SIGNATURE "Verify signature of a dependency during its dynamic run-time linking" ON)

Reviewer: I personally prefer the current name :)

> option(TBB_VERIFY_DEPENDENCY_SIGNATURE "Verify signature of a dependency during its dynamic run-time linking" ON)

Author: Agreed to not change the user-facing option, but amend the internal.

    include(${FILE_WITH_EXTRA_TARGETS})
endforeach()

if (NOT TBB_VERIFY_DEPENDENCY_SIGNATURE)
    message(WARNING
        "Dependency signature verification during dynamic run-time linking is disabled. This may "
        "lead to security vulnerabilities. It is recommended to leave it enabled in production "

>         "environment. Unset TBB_VERIFY_DEPENDENCY_SIGNATURE or set it ON to enable it again.")

Reviewer 2: Will unsetting TBB_VERIFY_DEPENDENCY_SIGNATURE not disable it instead? Did you mean "Set TBB_VERIFY_DEPENDENCY_SIGNATURE ..."

Author: I clarified the messaging.

    include(${FILE_WITH_EXTRA_TARGETS})
endforeach()

if (NOT TBB_VERIFY_DEPENDENCY_SIGNATURE AND WIN32)
    message(WARNING "Verification of a dependency signature during dynamic "
        "run-time linking is explicitly disabled. This may lead to security "
        "vulnerabilities. It is recommended to set "
        "TBB_VERIFY_DEPENDENCY_SIGNATURE to ON (default value)")

> endif()

Author: Consider adding `else` branch to show that binary signatures are enabled so that users are aware about potential issues. Something like:
```cmake
message(STATUS "oneTBB is configured to verify signatures of dynamic dependencies loaded during run-time.")
```

>         "This may lead to security vulnerabilities. It is recommended to leave it enabled by default"
>         " or set TBB_VERIFY_DEPENDENCY_SIGNATURE ON to enable it explicitly.")

Reviewer: What about making it shorter a little bit?
```suggestion
        "This may lead to security vulnerabilities. It is recommended to set "
        "TBB_VERIFY_DEPENDENCY_SIGNATURE to ON (default value)")
```

Author: Updated.
